hoomd/hpmc/UpdaterShape.h

```diff
@@ -0,0 +1,503 @@
+        return m_step_size[id];
+        }
+
+    inline void setStepSize(std::string name, Scalar d)
+        {
+        unsigned int id = this->m_pdata->getTypeByName(name);
+        m_step_size[id] = d;
+        }
+
+    void countParticlesPerType();
+
+    private:
+    std::shared_ptr<IntegratorHPMCMono<Shape>> m_mc; // hpmc particle integrator
+    std::shared_ptr<ShapeMoveBase<Shape>>
+        m_move_function;        // shape move function to apply in the updater
+    unsigned int m_type_select; // number of particle types to update in each move
+    unsigned int m_nsweeps;     // number of sweeps to run the updater each time it is called
+    bool m_pretend;             // whether or not to pretend or actually perform shape move
+    bool m_multi_phase;         // whether or not the simulation is multi-phase
+    unsigned int m_num_phase;   // number of phases in a multi-phase simulation
+    int m_global_partition;     // number of MPI partitions
+    GPUArray<Scalar>
+        m_determinant_inertia_tensor; // determinant of the shape's moment of inertia tensor
+    GPUArray<unsigned int> m_ntypes;  // number of particle types in the simulation
+    bool m_initialized;               // whether or not the updater has been initialized
+    std::vector<Scalar> m_step_size;  // shape move stepsize
+    std::vector<unsigned int> m_count_accepted; // number of accepted updater moves
+    std::vector<unsigned int> m_count_total;    // number of attempted updater moves
+    std::vector<unsigned int>
+        m_box_accepted; // number of accepted moves between boxes in multi-phase simulations
+    std::vector<unsigned int>
+        m_box_total; // number of attempted moves between boxes in multi-phase simulations
+    detail::UpdateOrder m_update_order; // order of particle types to apply the updater to
+    unsigned int m_instance = 0;        //!< Unique ID for RNG seeding
+    };
+
+template<class Shape>
+UpdaterShape<Shape>::UpdaterShape(std::shared_ptr<SystemDefinition> sysdef,
+                                  std::shared_ptr<Trigger> trigger,
+                                  std::shared_ptr<IntegratorHPMCMono<Shape>> mc,
+                                  std::shared_ptr<ShapeMoveBase<Shape>> move)
+    : Updater(sysdef, trigger), m_mc(mc), m_move_function(move), m_global_partition(0),
+      m_determinant_inertia_tensor(m_pdata->getNTypes(), m_exec_conf),
+      m_ntypes(m_pdata->getNTypes(), m_exec_conf), m_initialized(false)
+    {
+    m_step_size.resize(m_pdata->getNTypes(), 0);
+    m_count_accepted.resize(m_pdata->getNTypes(), 0);
+    m_count_total.resize(m_pdata->getNTypes(), 0);
+    m_box_accepted.resize(m_pdata->getNTypes(), 0);
+    m_box_total.resize(m_pdata->getNTypes(), 0);
+    initializeDeterminatsInertiaTensor();
+    countParticlesPerType();
+
+    if (m_multi_phase)
+        {
+#ifdef ENABLE_MPI
+        MPI_Comm_rank(MPI_COMM_WORLD, &m_global_partition);
+        assert(m_global_partition < 2);
```

Comment:
This assert should not be necessary. If we don't use the world comm below, but the communicator used by m_exec_conf.